pkg/identity/service.go

```diff
@@ -160,9 +160,12 @@ func checkCSR(csr *x509.CertificateRequest, identity string) error {
 		return fmt.Errorf("CSR name does not match requested identity: csr=%s; req=%s", csr.DNSNames[0], identity)
 	}
 
-	if csr.Subject.CommonName != "" {
-		return errors.New("CommonName must be empty")
+	switch csr.Subject.CommonName {
+	case "", identity:
```

Comment:
curious: should we now always expect the `csr.Subject.CommonName` to equal `identity.(NS).cluster.local`? is empty string valid going forward?

Reply:
We need to continue to support CSRs generated by prior versions or the upgrade path will not be clean.